packages/hooks/src/useTabsEventHandlers.js

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+
+function mathMod(a, b) {
+  return ((a % b) + b) % b;
+}
+function isObject(value) {
+  return !!value && typeof value === 'object';
+}
+
+export default function useTabsEventHandlers(tab, tabs, updaterFn, active, options = {}) {
+  const { customFindFn } = options;
+  if ((typeof tab !== 'string' && !isObject(tab)) || (isObject(tab) && !('name' in tab))) {
+    throw new Error('useTabsEventHandler requires tabs to be strings or objects with a name property');
+  }
+
+  if ((isObject(tab) || isObject(active)) && typeof customFindFn !== 'function') {
+    throw new Error('when using tab objects a custom find function is required');
```

Comment:
Similar to previous pr i am interested in feedback on this strategy of throwing errors as opposed to warning and using a sensible default like if(isObject(tab) && typeof customFindFn === 'undefined' {customFindFn = (tabs, active) => tabs.find(tab => tab.name === active.name}